mathics/builtin/box/graphics.py

```diff
@@ -798,12 +811,18 @@ def add_element(element):
                             ),
                         ]
                     )
+
+                    tick_value = 10**x if is_logscale else x
```

Comment:
Here is where better labels for logplots would be added.

Reply:
@mmatera Yes, I noticed this. Somewhere in a side comment I think I (or intended to) methion that this was a weakness. 

Also weak is robustness when a bad function is given, for example instead of `Sin[]` try `PartitionQ[]` which we don't implement. Or more generally when a single value can't be evaluated. 

Or more generally, it would have been good to write unit tests for the split out functions, and in also for error conditions. Or start adding docstrings for all of those internal functions that exist.

Please, please any of this interests you by all means continue here!

But this commit is getting large. There is too much to to do continuing here. (Every time we look behind some existing code there is way to much to try to redo, so at least for me, I need to break it up into little chunks and do as a number of PRs.)

But, right now I said that the main focus was thinking about how to get a release out. So we can then continue with the more serious stuff. 

The addition of plot routines like this I had mentioned _months_ ago, but because every week I was looking the more import deeper issues, I never had a chance to think about looking at something that is conceptually simpler.

Reply:
I just meant to fix it if it were a trivial change, or just add a comment to the TODO list. Again, I am OK with merging this just as it is right now.